libs/packages/sam-formly/src/lib/formly-filters/sds-filters.component.ts

```diff
@@ -99,39 +101,27 @@ export class SdsFiltersComponent implements OnInit {
     private cdr: ChangeDetectorRef,
     private router: Router,
     private route: ActivatedRoute,
+    @Optional()
+    private filterUpdateModelService: SDSFormlyUpdateModelService,
     library: FaIconLibrary
   ) {
     library.addIconPacks(fas, sds);
   }
 
-  @HostListener('window:popstate', ['$event'])
-  onpopstate(event) {
-    const queryString = window.location.search.substring(1);
-    const params = this.getUrlParams(queryString);
-    const updatedFormValue = this.overwrite(
-      this.form.getRawValue(),
-      this.convertToModel(params)
-    );
-    this.form.setValue(updatedFormValue);
-    this.updateChange(updatedFormValue);
-  }
   ngOnInit(): void {
-    if (this.isHistoryEnable) {
-      if (this._isEmpty(this.form.getRawValue())) {
-        const queryString = window.location.search.substring(1);
-        const params: any = this.getUrlParams(queryString);
-        const paramModel: any = this.convertToModel(params);
-        this.checkForHide();
-        setTimeout(() => {
-          this.form.patchValue({
-            ...this.model,
-            ...paramModel.sfm,
+    if (this.filterUpdateModelService) {
+      this.filterUpdateModelService.filterModel.subscribe((filter) => {
+        if (filter) {
+          const updatedFormValue = this.overwrite(
+            this.form.getRawValue(),
+            filter
+          );
+          setTimeout(() => {
+            this.form.patchValue(updatedFormValue);
           });
-        });
-        this.cdr.detectChanges();
-      }
-    } else if (this.model) {
-      this.checkForHide();
+        }
+      });
+      this.cdr.detectChanges();
```

Comment:
Could you also add in the hide check here? It overwrites the hide check that's done to display hidden filters with values.

Reply:
updated